### ISSUE TITLE
R2 is missing interior and posterior comments 

### ISSUE THREAD

**Issue by shiftre** [radare/radare2#3552]
Interior and Posterior comments allow to make the navigation and separation of function calling and parameters easier 

They also allow to add global comments about the current function being analyzed, unlike regular comments which are put on the side or repeatable comments which are special condition of regular comments.

They are referenced in IDA as Shift-Ins and Ins keyboard shortcuts.

**Comment by radare** [comment]
You mean asm.cmtright?

**Comment by crowell** [comment]
@radare yes, but for individual comments can be useful, that's a global var

**Comment by radare** [comment]
Yep. We can add a per-comment option to toggle this. This can be easily implemented on top of ah.. But would be desirable to be in C*

Also.. What do in case one option conflicts the other?

**Comment by shiftre** [comment]
How would they collide ?

**Comment by radare** [comment]
Now comments doesnt specify any position or type. They are just comments. The asm.cmtright option allows to choose where to render them.

The idea would be to allow to specify where to print any specific comment. This option will override the asm.cmtright variable.

Does it looks good for you?

**Comment by shiftre** [comment]
Does that mean that all the comments would be rendered this way if I choose this option ? If this is the case, then no .

I want to be able to have an option to use Interior, Exterior and Regular comments altogether 

For ex

anterior comment


; Attributes: bp-based frame

; int __cdecl main(int argc, const char **argv, const char **envp)
public main
main proc near

argc= dword ptr  8
argv= dword ptr  0Ch
envp= dword ptr  10h

push    ebp

posterior comment
mov     ebp, esp        ; regular comment


This one is taken from IDA .

**Comment by ret2libc** [comment]
This issue has been moved from radareorg/radare2 to radareorg/ideas as we are trying to clean our backlog and this issue has probably been created a long while ago. This is an effort to help contributors understand what are the actionable items they can work on, prioritize issues better and help users find active/duplicated issues more easily. If this is not an enhancement/improvement/general idea but a bug, feel free to ask for re-transfer to main repo. Thanks for your understanding and contribution with this issue.
